client/components/search-form.jsx

```diff
@@ -34,6 +33,18 @@ export default class SearchForm extends React.Component {
   }
 
   render() {
+    let restaurant;
+    if (!this.state.restaurant) {
+      restaurant = '';
+    } else {
+      restaurant = this.state.restaurant;
+    }
+    let location;
+    if (!this.state.location) {
+      location = '';
+    } else {
+      location = this.state.location;
+    }
```

Comment:
I don't think any of this is necessary, just use `this.state.(restaurant|location)` directly as they are initially empty strings.